$OpenBSD: patch-contrib_asdf-module_mk,v 1.2 2013/09/15 16:34:00 jca Exp $

Fix 'all' target to allow building without running tests.

Don't copy every single file when installing the contribs, only the
ones that are actually needed to load the system.

--- contrib/asdf-module.mk.orig	Sun Jun  2 15:12:39 2013
+++ contrib/asdf-module.mk	Sun Sep 15 14:11:33 2013
@@ -27,7 +27,8 @@ endif
 
 export CC SBCL EXTRA_CFLAGS EXTRA_LDFLAGS
 
-all: $(EXTRA_ALL_TARGETS)
+all: $(EXTRA_ALL_TARGETS) $(SYSTEM).fasl
+$(SYSTEM).fasl:
 	$(MAKE) -C ../asdf
 	$(SBCL) --eval '(defvar *system* "$(SYSTEM)")' --load ../asdf-stub.lisp --eval '(exit)'
 
@@ -40,5 +41,4 @@ test: all
 # KLUDGE: There seems to be no portable way to tell tar to not to
 # preserve owner, so chown after installing for the current user.
 install: $(EXTRA_INSTALL_TARGETS)
-	tar cf - . | ( cd "$(BUILD_ROOT)$(INSTALL_DIR)" && tar xpvf - )
-	find "$(BUILD_ROOT)$(INSTALL_DIR)" -exec chown `id -u`:`id -g` {} \;
+	cp -p $(SYSTEM).asd *.lisp *.fasl "$(BUILD_ROOT)$(INSTALL_DIR)"
